### DISCUSSION TITLE
OAuth Token expires_in time

### DISCUSSION THREAD

**Discussion by priyam1103** [etsy/open-api#996] [Q&A]
As mentioned in the documentation requesting a refresh OAuth Token gives the access token which is valid for 86400 seconds i.e. 24 hours but when I try to actually use this service it gives a token which is valid only for 3600 seconds i.e. 1 hour. Is this a known issue or I am doing something wrong.

This is my request

`https://api.etsy.com/v3/public/oauth/token?grant_type=refresh_token&client_id= 1aa2bb33c44d55eeeeee6fff&refresh_token=12345678.O1zLuwveeKjpIqCQFfmR-PaMMpBmagH6DljRAkK9qt05OtRKiANJOyZlMx3WQ_o2FdComQGuoiAWy3dxyGI4Ke_76PR`

Please help me on this.

**Comment by Gareth064** [comment]
Have a look at my thread here https://github.com/etsy/open-api/discussions/542

The first access_token you get from the user granting your app permissions to their store is only valid for 1hr
The refresh_token is valid for 90 days

The 24hrs you are reading in the docs is the expiration time mentioned in the section for getting a refresh token. This part of the documentation is confusing as it suggests that when you use a refresh token to get a new access token, then that new access token is valid for 24 hours.

Probably something to email the devs about to get confirmation on. Maybe the docs are worng.

@nateswart can you comment on this if you are about?
In this part of the docs, the example says the access token you get back after using the refresh token, has a life of 24 hours and not 1 hour like the very first access token. is that correct?
![image](https://github.com/etsy/open-api/assets/22307223/fd5eb6f3-0900-44f6-b5bd-767074224d98)
